slatedb/src/tablestore.rs

```diff
@@ -516,6 +503,48 @@ impl TableStore {
     }
 }
 
+async fn write_sst_with_retry(
+    object_store: Arc<dyn ObjectStore>,
+    id: &SsTableId,
+    data: &Bytes,
+    path: &Path,
+) -> Result<(), SlateDBError> {
+    let retry_strategy = ExponentialFactorBackoff::from_millis(10, 2.0)
+        .map(jitter)
+        .take(3);
+
+    Retry::spawn(retry_strategy, || async {
+        object_store
+            .put_opts(path, data.clone().into(), PutOptions::from(PutMode::Create))
+            .await
+            .map_err(|e| match e {
+                ObjectStoreError::AlreadyExists { .. } => match id {
+                    SsTableId::Wal(_) => {
+                        debug!("path already exists [path={}]", path);
+                        RetryError::permanent(SlateDBError::Fenced)
+                    }
+                    SsTableId::Compacted(_) => RetryError::permanent(e.into()),
+                },
+                e if is_timeout(&e) => RetryError::transient(e.into()),
+                _ => RetryError::permanent(e.into()),
+            })
+    })
+    .await?;
```

Comment:
Nit: can we simply end with `.await` instead of manually setting `Ok(())` at the end? Not sure what the anonymous function returns.